front-end/src/app/task.service.ts

```diff
@@ -1,28 +1,31 @@
 import { Injectable } from '@angular/core';
 import { Task } from 'src/model/Task';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Observable, firstValueFrom, lastValueFrom } from 'rxjs';
+import { TmplAstForLoopBlock } from '@angular/compiler';
 
 @Injectable({
   providedIn: 'root',
 })
 export class TaskService {
-  readonly url = 'http://localhost:3000/tasks';
+  readonly url = 'http://127.0.0.1:3000/tasks';
   db: Task[] = [];
 
-  constructor() {}
+  constructor(private http: HttpClient) {}
 
-  async getTasks(): Promise<Task[]> {
-    /*const data = await fetch(this.url);
-    return (await data.json()) ?? []; TODO: HA LESZ BACKEND AKKOR EZT HASZNÁLJUK!*/
-    return new Promise((resolve, reject) => {
-      resolve(this.db);
-    });
+  async getTasks(): Promise<any> {
+    const data = this.http.get(this.url);
+    const lastData = await lastValueFrom(data);
```

Comment:
try to be consistent with the usage of Promises and Observables in the service. You could use observable here as well

Reply:
If I try to use Observable type there's an error because async functions must be Promise type. I tried using Promise<Observable<any>> but it also causes trouble in the function and in app.component.

Reply:
It causes trouble imagining `Promise<Observable>` as well. :) You should use either one, but never both. This is a deeper topic, so let's dive in.

***

With **Promises** you can use 2 approaches to get the value of the promise, both will work:
- **using the `then` function**: it is not compulsory to use the `async` keyword for the function that returns a promise, therefore if you don't use `async`, you will still be able to get the value using the `.then()` function
- **async/await**: if you use the `async` keyword, you can still use the `.then()` function, but you can also use `await`

comparison:
```javascript
// using 'then'
// service function
getTasks(): Promise<Task[]> {
  // ...
}
// component code
this.taskService.getTasks().then(tasks => {
  tasks.foreach(task => {
    // ...
  })
})


// using 'async/await'
// service function
async getTasks(): Promise<Task[]> {
  // ...
}
// component code
const tasks = await this.taskService.getTasks();
tasks.foreach(task => {
  // ...
})
```

Because both work, it is a personal choice which one you use. _Code readability is also a factor, but it's beside the point._
In your code you are actually mixing the 2 promise forms. You use the `async` keyword, but instead of using the `await` you are using the `.then()` function.

***

With **Observables**:
- you cannot use  `async` and `await`
- you can use a `subscribe` function, which is similar to the promises' `then` function

I suggest that you first modify your code so you use only the `then` promise approach. If that is working, you can move on to changing the return type to Observable.